datasource/src/main/scala/quasar/physical/mongo/MongoConfig.scala

```diff
@@ -38,9 +40,21 @@ final case class MongoConfig(
       case Some(cn) => Some(Collection(db, cn))
     }}
   }
+
+  def withBatchSize(size: Int): MongoConfig =
+    this.copy(batchSize = size)
+  def withPushdown(level: PushdownLevel): MongoConfig =
+    this.copy(pushdownLevel = level)
+  def viaTunnel(cfg: TunnelConfig): MongoConfig =
+    this.copy(tunnelConfig = Some(cfg))
+  def useSSL(cfg: SSLConfig): MongoConfig =
+    this.copy(sslConfig = Some(cfg))
```

Comment:
Personally I like standard naming better here, ie `viaTunnel` -> `withTunnelConfig` and `useSSL` -> `withSSLConfig` but feel free to leave it like this if you prefer this

Reply:
I used `viaTunnel` and `useSSL` not `withTunnelConfig` because they aren't in line with `withBatchSize` actually. `withSSLConfig` should take `Option[SSLConfig]` :)

Reply:
RIght yeah, for me it's more that `withSomething` signals that it's part of builder pattern, the types don't need to line up exactly. 

But that can very well be just me - feel free to leave as is :)